File: ipcservice/src/main/java/com/android/lmc/ipcservice/MainActivity.java
package com.android.lmc.ipcservice;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    static {
        System.loadLibrary("hello");
    }

    public native String sayHello();

    @Override
    public void onResume(){
        super.onResume();
        Toast.makeText(MainActivity.this, sayHello(), Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/android/lmc/MainActivity.kt
package com.android.lmc

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.android.lmc.activity.CrashActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        CrashActivity.enterActivity(this)
    }
}
